File: design-interface/python-ref/wrap-output.py
import sys
import codecs
import rot13_codec  # noqa: F401  (registers the "my-rot13" codec)

with open("input-hello.txt", "rb") as f:
    plain = f.read()

writer = codecs.getwriter("my-rot13")(sys.stdout.buffer)
writer.write(plain)

File: design-interface/rust-twin-v1/src/bin/wrap_output.rs
//! Twin of `wrap-output.py`: wrap a byte output stream (stdout) with a codec
//! to get an encoding output stream.

use std::io::Write;

use rust_twin::{Rot13, StreamWriter};

fn main() -> std::io::Result<()> {
    let plain = std::fs::read(concat!(env!("CARGO_MANIFEST_DIR"), "/input-hello.txt"))?;
    let mut writer = StreamWriter::new(std::io::stdout().lock(), Rot13);
    writer.write_all(&plain)?;
    let _stdout = writer.finish()?;
    Ok(())
}
